core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -78,11 +78,11 @@ object ByteBufferMessageSet {
   }
 
   /** Deep iterator that decompresses the message sets and adjusts timestamp and offset if needed. */
-  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): Iterator[MessageAndOffset] = {
+  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): CloseableIterator[MessageAndOffset] = {
```

Comment:
@ijuma I *think* old clients don't rely on this method - they should be getting a `ConsumerIterator` in the high level api and a `FetchResponse` in the `SimpleConsumer`. As such it isn't public, unless I'm mistaken.